[clang][static analyzer]alpha.deadcode.UnreachableCode false positive on cxxTryStmt

```cpp
  class BaseException {};
  
  class DerivedException : public BaseException {};
  
  void foo();
  
  void f4() {
    try {
      foo();
    } catch (BaseException &b) {
    } catch (DerivedException &d) {
    }
  }
```

When I run ```clang-tidy --allow-enabling-analyzer-alpha-checkers -checks=-*,clang-analyzer-alpha.deadcode.UnreachableCode test.cpp```, the result like as following:

```bash
test.cpp:8:3: warning: This statement is never executed [clang-analyzer-alpha.deadcode.UnreachableCode]
    8 |   try {
      |   ^~~~~
    9 |     foo();
      |     ~~~~~~
   10 |   } catch (BaseException &b) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   11 |   } catch (DerivedException &d) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   12 |   }
      |   ~
test.cpp:8:3: note: This statement is never executed
    8 |   try {
      |   ^~~~~
    9 |     foo();
      |     ~~~~~~
   10 |   } catch (BaseException &b) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   11 |   } catch (DerivedException &d) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   12 |   }
      |   ~

```

I think this is false positive. If not , I am not clear about reason.

--- a/clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp
+++ b/clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp
@@ -159,6 +159,8 @@ void UnreachableCodeChecker::checkEndAnalysis(ExplodedGraph &G,
       SL = DL.asLocation();
       if (SR.isInvalid() || !SL.isValid())
         continue;
+      if (isa<CXXTryStmt>(S))
+        continue;
     }
     else
       continue;

--- a/clang/test/Analysis/unreachable-code-exceptions.cpp
+++ b/clang/test/Analysis/unreachable-code-exceptions.cpp
@@ -4,9 +4,10 @@
 
 void foo();
 
-void f4() {
-  try {
+void fp_90162() {
+  try { // no-warning: The TryStmt shouldn't be unreachable.
     foo();
   } catch (int) {
+    foo(); // We assume that catch handlers are reachable.
   }
 }

--- a/clang/test/Analysis/unreachable-code-exceptions.cpp
+++ b/clang/test/Analysis/unreachable-code-exceptions.cpp
@@ -1,0 +1,12 @@
+// RUN: %clang_analyze_cc1 -verify %s -fcxx-exceptions -fexceptions -analyzer-checker=core -analyzer-checker=alpha.deadcode.UnreachableCode
+
+// expected-no-diagnostics
+
+void foo();
+
+void f4() {
+  try {
+    foo();
+  } catch (int) {
+  }
+}

--- a/clang/test/Analysis/unreachable-code-exceptions.cpp
+++ b/clang/test/Analysis/unreachable-code-exceptions.cpp
@@ -1,4 +1,4 @@
-// RUN: %clang_analyze_cc1 -verify %s -fcxx-exceptions -fexceptions -analyzer-checker=core -analyzer-checker=alpha.deadcode.UnreachableCode
+// RUN: %clang_analyze_cc1 -verify %s -fcxx-exceptions -fexceptions -analyzer-checker=core,alpha.deadcode.UnreachableCode
 
 // expected-no-diagnostics
 

